doc/port_auto_neg/port-auto-negotiation-design.md

```diff
@@ -84,8 +92,13 @@ Currently, SAI already defines a few port attributes to support port auto negoti
 1. SAI implementation must return error code if any of the auto negotiation related attribute is not supported, swss and syncd must not crash.
 2. SAI implementation must keep backward compatible. As long as swss and SAI keep backward compatible, user need not change anything after this feature is enabled in SONiC.
 3. If autoneg is enabled and adv_speeds is not configured or empty, SAI must advertise it with all supported speeds.
-4. If autoneg is enabled and adv_interface_types is not configured or empty, SAI must advertise it with all supported interface types.
-5. If autoneg is disabled and interface_type is not configured, SAI must use SAI_PORT_INTERFACE_TYPE_NONE.
+4. If autoneg is enabled and adv_speeds is specified, SAI must advertise only the speeds those are supported by the switch silicon. Hence the operational advertisement could be a subset of the speeds specified in adv_speeds. SAI should return errors if none of desirable advertised speeds is valid.
+5. If autoneg is enabled and adv_interface_types is not configured or empty, SAI must advertise it with all supported interface types.
+6. If autoneg is enabled and adv_interface_types is specified, SAI must advertise only the interface types those are valid to the attched transceiver and supported by the switch silicon. Hence the operational advertisement could be a subset of the interface types specified in adv_interface_types. SAI should return errors if none of the desirable advertised interface types is valid.
+7. If autoneg is enabled, the administrative port speed updates via SAI_PORT_ATTR_SPEED should disable the autoneg.
```

Comment:
As discussed in the review remove the point 7. It should be replaced by a check in sonic-utilities to block speed configuration when autoneg is enabled.

Reply:
This check should also be done for other attributes that are exchanged via autoneg. Some of them I can name are FEC, interface type.

Reply:
Copy that, the HLD will be updated as follow
```
7. If autoneg is enabled, the administrative port speed updates (e.g. SAI_PORT_ATTR_SPEED) should not alter the autoneg config and operational states. The sonic-utilities should also block speed configuration when autoneg is enabled.
8. If autoneg is enabled, the administrative port FEC updates (e.g. SAI_PORT_ATTR_FEC_MODE, SAI_PORT_ATTR_FEC_MODE_EXTENDED) should not alter the autoneg config and operational states. The sonic-utilities should also block FEC configuration when autoneg is enabled.

```